src/Chamilo/UserBundle/Entity/User.php

```diff
@@ -2469,4 +2483,107 @@ public function getSessionAsGeneralCoach()
+        $className = get_class($this);
+        $metadata = $em->getClassMetadata($className);
+        $maxRecursionDepth = 5;
+
+        $data = array();
+
+        foreach ($metadata->fieldMappings as $field => $mapping) {
+            $value = $metadata->reflFields[$field]->getValue($this);
+            $field = Inflector::tableize($field);
+            if ($value instanceof \Datetime) {
+                $data[$field] = (array)$value;
+            } elseif (is_object($value)) {
+                $data[$field] = (string)$value;
+            } else {
+                $data[$field] = $value;
+            }
+        }
+
+        foreach ($metadata->associationMappings as $field => $mapping) {
+            $key = Inflector::tableize($field);
+            if ($mapping['isCascadeDetach']) {
+                $data[$key] = $metadata->reflFields[$field]->getValue($this);
+                if (null !== $data[$key]) {
+                    $data[$key] = $this->_toArray($data[$key], $recursionDepth + 1);
+                }
+            } elseif ($mapping['isOwningSide'] && $mapping['type'] & ORMMeta::TO_ONE) {
+                if (null !== $metadata->reflFields[$field]->getValue($this)) {
+                    if ($recursionDepth < $maxRecursionDepth) {
+                        $recursionDepth++;
+                        $data[$key] = $this->_toArray(
+                            $metadata->reflFields[$field]->getValue($this),
+                            $recursionDepth
+                        );
+                        $recursionDepth--;
+                    } else {
+                        $data[$key] = $em->getUnitOfWork()
+                            ->getEntityIdentifier(
+                                $metadata->reflFields[$field]
+                                    ->getValue($this)
+                                );
+                    }
+                } else {
+                    // In some case the relationship may not exist, but we want
+                    // to know about it
+                    $data[$key] = null;
+                }
+            }
+        }
+        return $data;
+    }
+
+    /**
+     * Serialize the whole entity to an array
+     * @param EntityManager $em
+     * @return array $values
+     */
+    public function getPersonalData($em)
+    {
+        $d = $this->_toArray($em);
+        foreach ($d as $key => $value) {
+            switch ($key) {
+                case 'password':
+                    $d[$key] = get_lang('EncryptedData');
+                    break;
+                case 'salt':
+                    $d[$key] = get_lang('RandomData');
+                    break;
+            }
+            if (empty($value)) {
+                $d[$key] = get_lang('NoData');
+            }
+
+        }
+        return $d;
+    }
+
+    /**
+     * Get last login (used only if user.last_login is empty)
+     * @return \DateTime Last login date for this user
+     */
+    public function getExtendedLastLogin()
```

Comment:
api_get_local_time() cannot be here, this depends of Chamilo, better to make the conversion outside the entity.
It should just check UTC times.